app/controllers/attendances_controller.rb

```diff
@@ -0,0 +1,58 @@
+class AttendancesController < ApplicationController
+  before_action :set_user, :set_group, :authenticate_user!
+  before_action :set_attendance, only: [:show]
+
+  def index
+    if params[:date].present?
+      @attendances = @group.attendances.where(date: params[:date])
+      redirect_to teacher_group_url(@user, @group), alert: "Not attendances for this day." if @attendances.count.zero?
```

Comment:
I don't think this should be here, in attendance index we should display the attendances for the selected date or this message, don't do a redirect for that